app/src/main/java/ch/blinkenlights/android/vanilla/CoverView.java

```diff
@@ -202,6 +202,11 @@ private void querySongsInternal() {
 			Song song = songs[i];
 			if (mBitmapBucket.getSong(i) != song) {
 				mHandler.sendMessage(mHandler.obtainMessage(MSG_SET_BITMAP, i, 0, song));
+			} else if (force) {
+				// force bitmap invalidation. CoverCache must be already invalidated
```

Comment:
> but i really do not like the force parameter: there should be a nicer way to solve this.

`Song` objects have a very bad `Comparator` and don't even have the `equals` overridden. They aren't immutable and we're comparing them by reference in CoverView. When I tried to fix this it blew up all over the place. The `force` parameter was the only thing that worked without shaking the interfaces too much.

> maybe it would be simpler to just have new observer for cache invalidation where the cache holders subscribe to - and playback service would then publish invalidation hints.

This will make things even more complex. We'll have one more observer-notifier to care about and it won't even apply changes to title/artist fields in FullPlaybackView.